core/src/main/scala/cats/Foldable.scala

```diff
@@ -152,6 +152,26 @@ import simulacrum.typeclass
   def size[A](fa: F[A]): Long = foldMap(fa)(_ => 1)
 
   /**
+    * Get the element at the index of the `Foldable`.
+    *
+    * The default implementation of this is based on `foldLeft`, and thus will
+    * always fold across the entire structure.
+    */
+  def get[A](fa: F[A])(idx: Long): Option[A] =
+    foldLeft(fa, (0L, Option.empty[A])) { (b, a) =>
+      b._1 match {
+        case i if i == idx  => (i + 1, Some(a))
+        case i if i < idx => (i + 1, None)
+        case _ => b
+      }
+    }._2
+
+  /**
+    * Get the element at the index of the `Foldable`
+    */
+  def apply[A](fa: F[A])(idx: Long): A = get(fa)(idx).get
```

Comment:
@yilinwei I think `.get` is a good method to have.

I think I agree with others that `apply` is a bit harder to justify. If we have a concrete type (e.g. `Vector`) then the caller can use the vector's own `.apply` method to index into it. rather than needing to rely on `Foldable[F].apply`.

When dealing with a totally generic type `F[A]` with `Foldable[F]` I think it's harder to be sure that a given index will be present.